module/cloud-services/files/helper.js

```diff
@@ -0,0 +1,155 @@
+
+				let result = {
+					[payloadIds[0]]: {},
+				}
+
+				for (let pointerToPayload = 0; pointerToPayload < payloadIds.length; pointerToPayload++) {
+					let payloadId = payloadIds[pointerToPayload]
+					// Generate unique folderPath to all the file names in payloadData
+					let folderPath = 'project/' + payloadId + '/' + userId + '/' + UTILS.generateUniqueId() + '/'
+					// Call preSignedUrls helper file to get the signedUrl
+					let imagePayload = await filesHelpers.preSignedUrls(
+						payloadData[payloadId].files,
+						bucketName,
+						cloudStorage,
+						folderPath,
+						typeof process.env.PRESIGNED_URL_EXPIRY_IN_SECONDS === 'string'
+							? parseInt(process.env.PRESIGNED_URL_EXPIRY_IN_SECONDS)
+							: process.env.PRESIGNED_URL_EXPIRY_IN_SECONDS, //expireIn PARAMS
+						'' //permission PARAMS
+					)
+
+					if (!imagePayload.success) {
+						return resolve({
+							status: HTTP_STATUS_CODE.bad_request.status,
+							message: CONSTANTS.apiResponses.FAILED_PRE_SIGNED_URL,
+							result: {},
+						})
+					}
+
+					if (!result[payloadId]) {
+						result[payloadId] = {}
+					}
+
+					result[payloadId]['files'] = imagePayload.result
+				}
+
+				return resolve({
+					message: CONSTANTS.apiResponses.URL_GENERATED,
+					data: result,
+				})
+			} catch (error) {
+				return reject(error)
+			}
+		})
+	}
+
+	/**
+	 * Get Downloadable URL from cloud.
+	 * @method
+	 * @name getDownloadableUrl
+	 * @param {Array} payloadData       - payload for files data.
+	 * @returns {JSON}                  - Response with status and message.
+	 */
+
+	static getDownloadableUrl(payloadData) {
+		return new Promise(async (resolve, reject) => {
+			try {
+				// if bucktType is private call preSignedUrls function with read permission
+				if (bucktType === CONSTANTS.common.PRIVATE) {
+					let downloadableUrl = await filesHelpers.preSignedUrls(
+						payloadData,
+						bucketName,
+						cloudStorage,
+						'',
+						typeof process.env.DOWNLOADABLE_URL_EXPIRY_IN_SECONDS === 'string'
+							? parseInt(process.env.DOWNLOADABLE_URL_EXPIRY_IN_SECONDS)
+							: process.env.DOWNLOADABLE_URL_EXPIRY_IN_SECONDS, //expireIn PARAMS
+						CONSTANTS.common.READ_PERMISSION, //permission PARAMS
+						true //true if filePath is passed
+					)
+
+					if (!downloadableUrl.success) {
+						return resolve({
+							status: HTTP_STATUS_CODE.bad_request.status,
+							message: CONSTANTS.apiResponses.FAILED_TO_CREATE_DOWNLOADABLEURL,
+							result: {},
+						})
+					}
+
+					return resolve({
+						message: CONSTANTS.apiResponses.CLOUD_SERVICE_SUCCESS_MESSAGE,
+						result: downloadableUrl.result,
+					})
+				}
+				// if bucketType is public go with the normal flow of calling getDownloadableUrl function
+				let downloadableUrl = await filesHelpers.getDownloadableUrl(
+					payloadData,
+					bucketName,
+					cloudStorage,
+					typeof process.env.DOWNLOADABLE_URL_EXPIRY_IN_SECONDS === 'string'
```

Comment:
@Prajwal17Tunerlabs Here also, we can avoid the extra if condition check

Reply:
removed the check